src/libraries/Common/src/System/Net/IPAddressParser.Common.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Numerics;
+
+namespace System.Net
+{
+    internal static partial class IPAddressParser<TChar>
+        where TChar : unmanaged, IBinaryInteger<TChar>
```

Comment:
Many members on IPAddressParser classes don't have to be generic (e.g. constants, anything on Uri ...).
I would avoid making the helper itself generic and only mark the necessary methods instead. That would also let you avoid having to specify the `TChar` at every call site as it would instead be inferred from the arguments.